`test_execution_state` is flaky on Windows and more often fails than not

## Description

Across multiple PRs and on the `main` branch this failure is evident:

```
FAILED tests/services/kernels/test_execution_state.py::test_execution_state - AssertionError: assert 'starting' == 'busy'
```

https://github.com/jupyter-server/jupyter_server/actions/runs/19349886439/job/55359089935

--- a/.github/workflows/python-tests.yml
+++ b/.github/workflows/python-tests.yml
@@ -49,7 +49,7 @@ jobs:
         run: hatch run test:nowarn || hatch -v run test:nowarn --lf
       - name: Run the tests on windows
         if: ${{ startsWith(matrix.os, 'windows') }}
-        run: hatch run cov:nowarn -s || hatch -v run cov:nowarn --lf
+        run: hatch run cov:nowarn -s -k test_execution_state || hatch -v run cov:nowarn --lf
       - uses: jupyterlab/maintainer-tools/.github/actions/upload-coverage@v1
 
   test_docs:

--- a/tests/services/kernels/test_execution_state.py
+++ b/tests/services/kernels/test_execution_state.py
@@ -17,7 +17,7 @@ POLL_INTERVAL = 1
 MINIMUM_CONSISTENT_COUNT = 4
 
 
-@pytest.mark.flaky(max_runs=5)
+@pytest.mark.flaky
 async def test_execution_state(jp_fetch, jp_ws_fetch):
     r = await jp_fetch("api", "kernels", method="POST", allow_nonstandard_methods=True)
     kernel = json.loads(r.body.decode())
@@ -53,6 +53,17 @@ async def test_execution_state(jp_fetch, jp_ws_fetch):
     )
     await poll_for_parent_message_status(kid, message_id, "busy", ws)
     es = await get_execution_state(kid, jp_fetch)
+
+    # kernels start slowly on Windows
+    max_startup_time = 60
+    started = time.time()
+    while es == "starting":
+        await asyncio.sleep(1)
+        elapsed = time.time() - started
+        if elapsed > max_startup_time:
+            raise ValueError(f"Kernel did not start up in {max_startup_time} seconds")
+        es = await get_execution_state(kid, jp_fetch)
+
     assert es == "busy"
 
     message_id_2 = uuid.uuid1().hex

--- a/tests/services/kernels/test_execution_state.py
+++ b/tests/services/kernels/test_execution_state.py
@@ -17,7 +17,7 @@ POLL_INTERVAL = 1
 MINIMUM_CONSISTENT_COUNT = 4
 
 
-@pytest.mark.flaky
+@pytest.mark.flaky(max_runs=3)
 async def test_execution_state(jp_fetch, jp_ws_fetch):
     r = await jp_fetch("api", "kernels", method="POST", allow_nonstandard_methods=True)
     kernel = json.loads(r.body.decode())

--- a/tests/services/kernels/test_execution_state.py
+++ b/tests/services/kernels/test_execution_state.py
@@ -17,7 +17,7 @@ POLL_INTERVAL = 1
 MINIMUM_CONSISTENT_COUNT = 4
 
 
-@pytest.mark.flaky(max_runs=3)
+@pytest.mark.flaky(max_runs=5)
 async def test_execution_state(jp_fetch, jp_ws_fetch):
     r = await jp_fetch("api", "kernels", method="POST", allow_nonstandard_methods=True)
     kernel = json.loads(r.body.decode())

